crates/red_knot_python_semantic/src/semantic_index/definition.rs

```diff
@@ -589,7 +657,27 @@ impl<'db> From<Option<Unpack<'db>>> for TargetKind<'db> {
 }
 
 #[derive(Clone, Debug)]
-#[allow(dead_code)]
+pub struct StarImportDefinitionKind {
+    node: AstNodeRef<ast::StmtImportFrom>,
+    symbol_id: ScopedSymbolId,
+    alias_index: usize,
```

Comment:
It's sort-of unfortunate that we need to keep track of what index the alias is in the names list for `*` imports, since it's invalid syntax to do something like `from foo import *, X` -- if `*` is present in the names list of a `StmtImportFrom` node, it must be the _only_ name in the list. However, if we _don't_ keep track of the index the `*` is in the names list, we end up panicking on various kinds of invalid syntax.

Reply:
Could we just search for the first alias with a * name? Or we could not  create a star import definition if the syntax is invalid

Reply:
Ohh of course, thanks! I'll make that change

Reply:
https://github.com/astral-sh/ruff/pull/16923/commits/a6f41525f2f0b9a175b6250a20c9319c72a40242